syntax = "proto3";

import "tensorflow/compiler/xla/xla_data.proto";

package xla;

// Backend specific HloInstruction config
message PoplarBackendConfig {

  // Instruction config for a fusion instruction
  message FusionConfig {
    //Convolutions
    Window window = 1;
    ConvolutionDimensionNumbers dimension_numbers = 2;
    int64 feature_group_count = 3;
    int64 batch_group_count = 4;
  };

  FusionConfig fusion_config = 1;

};
